packages/core/core.ts

```diff
@@ -84,18 +93,37 @@ export async function initFuzzing(options: Options) {
 				: new MemorySyncIdStrategy()
 		)
 	);
+	// Loads custom hook files and adds them to the hook manager.
 	await Promise.all(options.customHooks.map(ensureFilepath).map(importModule));
+	// Load built-in modules hooked by the `customHooks` and hooks them.
+	await registerHooksForBuiltinFunctions(hooking.hookManager);
+}
+
+async function registerHooksForBuiltinFunctions(
+	hookManager: hooking.HookManager
+) {
+	loadBugDetectors();
+	for (const builtinModule of builtinModules) {
+		for (const hook of hookManager.getMatchingHooks(builtinModule)) {
+			try {
+				await hooking.hookBuiltInFunction(hook);
```

Comment:
Add a comment about why the existing instrumentation approach isn't working ind why it's necessary to replace the build in functions.

Reply:
done